R/archive.R

```diff
@@ -196,572 +187,755 @@ next_after.Date <- function(x) x + 1L
-      #'   will be set to "custom".
-      #' @param other_keys Character vector specifying the names of variables in `x`
-      #'   that should be considered key variables (in the language of `data.table`)
-      #'   apart from "geo_value", "time_value", and "version".
-      #' @param additional_metadata List of additional metadata to attach to the
-      #'   `epi_archive` object. The metadata will have `geo_type` and `time_type`
-      #'   fields; named entries from the passed list or will be included as well.
-      #' @param compactify Optional; Boolean or `NULL`: should we remove rows that are
-      #'   considered redundant for the purposes of `epi_archive`'s built-in methods
-      #'   such as `as_of`? As these methods use the last version of each observation
-      #'   carried forward (LOCF) to interpolate between the version data provided,
-      #'   rows that don't change these LOCF results can potentially be omitted to
-      #'   save space while maintaining the same behavior (with the help of the
-      #'   `clobberable_versions_start` and `versions_end` fields in some edge cases).
-      #'   `TRUE` will remove these rows, `FALSE` will not, and missing or `NULL` will
-      #'   remove these rows and issue a warning. Generally, this can be set to
-      #'   `TRUE`, but if you directly inspect or edit the fields of the `epi_archive`
-      #'   such as its `DT`, or rely on redundant updates to achieve a certain
-      #'   behavior of the `ref_time_values` default in `epix_slide`, you will have to
-      #'   determine whether `compactify=TRUE` will produce the desired results. If
-      #'   compactification here is removing a large proportion of the rows, this may
-      #'   indicate a potential for space, time, or bandwidth savings upstream the
-      #'   data pipeline, e.g., by avoiding fetching, storing, or processing these
-      #'   rows of `x`.
-      #' @param clobberable_versions_start Optional; as in [`as_epi_archive`]
-      #' @param versions_end Optional; as in [`as_epi_archive`]
-      #' @return An `epi_archive` object.
-      #' @importFrom data.table as.data.table key setkeyv
-      #'
-      #' @details
-      #' Refer to the documentation for [as_epi_archive()] for more information
-      #' and examples of parameter names.
-      initialize = function(x, geo_type, time_type, other_keys,
-                            additional_metadata, compactify,
-                            clobberable_versions_start, versions_end) {
-        assert_data_frame(x)
-        if (!test_subset(c("geo_value", "time_value", "version"), names(x))) {
-          cli_abort(
-            "Columns `geo_value`, `time_value`, and `version` must be present in `x`."
-          )
-        }
-        if (anyMissing(x$version)) {
-          cli_abort("Column `version` must not contain missing values.")
-        }
+#' @name epi_archive
+epi_archive <- function(...) {
+  new_epi_archive(...)
+}
```

Comment:
suggest: delete this, but make sure `?epi_archive` still works.  People may expect that you can feed in column data in a manner similar to `data.frame`, `data.table`, `tibble`, etc., especially with the dots delegation being used.